hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/BulkInsertPartitioner.java

```diff
@@ -83,20 +84,28 @@ default Option<WriteHandleFactory> getWriteHandleFactory(int partitionId) {
 
   /*
    * If possible, we want to sort the data by partition path. Doing so will reduce the number of files written.
+   * Suffix the recordKeys as well if BULKINSERT_SUFFIX_RECORD_KEY_FOR_USER_DEFINED_SORT_COLUMNS is enabled, this reduces skew.
    * This will not change the desired sort order, it is just a performance improvement.
    **/
-  static String[] tryPrependPartitionPathColumns(String[] columnNames, HoodieWriteConfig config) {
+  static String[] tryPrependPartitionPathAndSuffixRecordKeyColumns(String[] columnNames, HoodieWriteConfig config) {
```

Comment:
Have you updated this PR with the latest changes?

Reply:
Updated.